### DISCUSSION TITLE
Do i need to cache connection? How to cache it?

### DISCUSSION THREAD

**Discussion by skorphil** [Automattic/mongoose#14425] [Q&A]
Hi, there are numerous articles over the web suggesting to cache mongoose.connection like so:
```js
let cached = global.mongoose

if (!cached) {
  cached = global.mongoose = { conn: null }
}
```
however i didn't find anything related on mongoose docs.

Is it considered good practice to cache connection (to avoid unnecessary connections)? And what is the conventions of doing this?

**Comment by skorphil** [comment]
as far as my research went, seems like mongoose already has https://mongoosejs.com/docs/connections.html#connection_pools so no need in additional logic to cache connections. Am i right?

**Reply by vkarpov15** [reply]
That pattern is specifically for serverless runtimes, like [AWS Lambda](https://mongoosejs.com/docs/lambda.html) and [Vercel Serverless Functions](https://mongoosejs.com/docs/nextjs.html). If you are running Mongoose on a server, there's no need to use that pattern.

**Reply by skorphil** [reply]
> That pattern is specifically for serverless runtimes, like [AWS Lambda](https://mongoosejs.com/docs/lambda.html) and [Vercel Serverless Functions](https://mongoosejs.com/docs/nextjs.html). If you are running Mongoose on a server, there's no need to use that pattern.

Thank you 
Can this pattern be replaced with `maxPoolSize: 1` for serverless functions or is it completely different concepts?

**Reply by vkarpov15** [reply]
Different concepts. This pattern is specifically for retaining connections between serverless function calls, so you don't need to recreate a connection to MongoDB every single time your serverless function runs. `maxPoolSize` doesn't help with retaining connections.
